TEST.md

```diff
@@ -0,0 +1,64 @@
+# Sourcegraph Helm Chart Test Guide
+
+This is a temporary test guide before we have more throughout automated integration tests.
+
+## Lint
+
+You shouldn't see any error.
+
+```sh
+helm lint charts/sourcegraph/.
+```
+
+## Unit testing
+
+We utilize [helm-unittest], a BDD styled unit test framework, to validate our helm chart.
+
+We currently do not have testing best practices or require unit tests for new changes, so add test cases at your best judgement if possible.
+
+You may check out our [existing test cases](https://github.com/sourcegraph/deploy-sourcegraph-helm/tree/main/charts/sourcegraph/tests) and helm-unittest [docs](https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md).
+
+## Manual testing
+
+Create a local cluster with [kind]. You may also use [k3s] or [minikube].
+
+```sh
+kind create cluster
+```
+
+You should have a `override.yaml` file ready to verify your changes.
+
+```yaml
+# Disable SC creation
+storageClass:
+  create: false
+  name: standard
+
+# Disable resources requests/limits
+sourcegraph:
+  localDevMode: true
+# More values to be added in order to test your change
+```
+
+It's a good idea to inspect the rendered manifest to catch things that look off.
```

Comment:
More recommendations:
- Perform a diff of the rendered helm manifests before and after your change
- Test both enabled and disabled toggles